frost-core/src/verifying_key.rs

```diff
@@ -37,12 +37,22 @@ where
     pub fn verify(&self, msg: &[u8], signature: &Signature<C>) -> Result<(), Error> {
         let c = crate::challenge::<C>(&signature.R, &self.element, msg);
 
+        self.verify_prehashed(signature, c)
+    }
+
+    /// Verify a purported `signature` with a pre-hashed [`Challenge`] made by this verification
+    /// key.
+    pub fn verify_prehashed(
```

Comment:
Does this need to be public (instead of `pub(crate`)? If yes, I'd suggest using some other name (e.g. `verify_challenge`). "prehashed" is a term used in RFC 8032 to signify something similar but different (passing a hash as the message) and may be confusing for users that are familiar with that RFC

Reply:
I'll rename, even if it's only used in `batch`

Reply:
Made `pub(crate)` and keeping name for now